presto-hive/src/test/java/com/facebook/presto/hive/TestOrcFileWriter.java

```diff
@@ -0,0 +1,145 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.common.ErrorCode;
+import com.facebook.presto.common.io.DataOutput;
+import com.facebook.presto.common.io.DataSink;
+import com.facebook.presto.orc.DefaultOrcWriterFlushPolicy;
+import com.facebook.presto.orc.OrcWriterOptions;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.common.ErrorType.EXTERNAL;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.hive.parquet.ParquetTester.HIVE_STORAGE_TIME_ZONE;
+import static com.facebook.presto.orc.DwrfEncryptionProvider.NO_ENCRYPTION;
+import static com.facebook.presto.orc.NoOpOrcWriterStats.NOOP_WRITER_STATS;
+import static com.facebook.presto.orc.OrcEncoding.ORC;
+import static com.facebook.presto.orc.metadata.CompressionKind.NONE;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+
+public class TestOrcFileWriter
+{
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Dummy PrestoException from mocked data sink instance")
+    public void testPrestoExceptionPropagation()
+    {
+        // This test is to verify that a PrestoException thrown by the underlying data sink implementation is propagated as is
+        OrcFileWriter orcFileWriter = new OrcFileWriter(
+                new MockDataSink(true),
+                () -> null,
+                ORC,
+                ImmutableList.of("test1"),
+                ImmutableList.of(VARCHAR),
+                NONE,
+                OrcWriterOptions.builder()
+                        .withFlushPolicy(DefaultOrcWriterFlushPolicy.builder()
+                                .withStripeMinSize(new DataSize(0, MEGABYTE))
+                                .withStripeMaxSize(new DataSize(32, MEGABYTE))
+                                .withStripeMaxRowCount(10)
+                                .build())
+                        .withRowGroupMaxRowCount(10_000)
+                        .withDictionaryMaxMemory(new DataSize(32, MEGABYTE))
+                        .build(),
+                new int[0],
+                ImmutableMap.of(),
+                HIVE_STORAGE_TIME_ZONE,
+                Optional.empty(),
+                null,
+                NOOP_WRITER_STATS,
+                NO_ENCRYPTION,
+                Optional.empty());
+
+        // Throws PrestoException with STORAGE_ERROR error code
+        orcFileWriter.commit();
+    }
+
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Error committing write to Hive.*")
```

Comment:
instead of just asserting on the exception message and type alone, we should catch it and assert on the cause as well so as to check if the cause is indeed the IOException.